container-chains/primitives/authorities-noting-inherent/src/tests.rs

```diff
@@ -0,0 +1,437 @@
+// Copyright (C) Moondance Labs Ltd.
+// This file is part of Tanssi.
+
+// Tanssi is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Tanssi is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Tanssi.  If not, see <http://www.gnu.org/licenses/>
+
+use {
+    super::*,
+    crate::ContainerChainAuthoritiesInherentData,
+    async_trait::async_trait,
+    cumulus_primitives_core::{
+        relay_chain::{
+            CommittedCandidateReceipt, HeadData, OccupiedCoreAssumption, SessionIndex, ValidatorId,
+        },
+        InboundDownwardMessage, InboundHrmpMessage, ParaId, PersistedValidationData,
+    },
+    cumulus_relay_chain_interface::{PHash, PHeader, RelayChainInterface, RelayChainResult},
+    futures::Stream,
+    polkadot_overseer::Handle,
+    sc_client_api::{StorageKey, StorageProvider},
+    sp_inherents::{InherentData, InherentDataProvider},
+    sp_state_machine::{prove_read, StorageValue},
+    std::{collections::BTreeMap, pin::Pin, sync::Arc},
+    substrate_test_runtime_client::{
+        ClientExt, DefaultTestClientBuilderExt, TestClient, TestClientBuilder, TestClientBuilderExt,
+    },
+    tc_orchestrator_chain_interface::{OrchestratorChainInterface, OrchestratorChainResult},
+    tp_core::{well_known_keys, Header as OrchestratorHeader},
+};
+
+#[derive(Clone)]
+struct DummyOrchestratorChainInterface {
+    orchestrator_client: Arc<TestClient>,
+}
+
+#[derive(Clone)]
+struct DummyRelayChainInterface {
+    relay_client: Arc<TestClient>,
+}
+const VALUE: &[u8] = b"hello world";
+
+impl DummyOrchestratorChainInterface {
+    fn new(session: u32) -> Self {
+        let builder = TestClientBuilder::new()
+            .add_extra_storage(
+                well_known_keys::SESSION_INDEX.to_vec(),
+                session.encode().to_vec(),
+            )
+            .add_extra_storage(
+                well_known_keys::authority_assignment_for_session(session).to_vec(),
+                VALUE.to_vec(),
+            );
```

Comment:
What's the purpose of this dummy storage? If I remove it all tests still pass.

Reply:
yes you are right this is not needed (since we dont verify against anything this storage). This can be null. Thes session one cannot be null thought bcause we need it to construct the second key